samza-rest/src/main/java/org/apache/samza/monitor/LocalStoreMonitor.java

```diff
@@ -81,25 +81,33 @@ public void monitor() throws Exception {
     for (JobInstance jobInstance : getHostAffinityEnabledJobs(localStoreDir)) {
       File jobDir = new File(localStoreDir,
                              String.format("%s-%s", jobInstance.getJobName(), jobInstance.getJobId()));
-      JobStatus jobStatus = jobsClient.getJobStatus(jobInstance);
-      for (Task task : jobsClient.getTasks(jobInstance)) {
-        for (String storeName : jobDir.list(DirectoryFileFilter.DIRECTORY)) {
-          LOG.info("Job: {} has the running status: {} with preferred host:  {}", jobInstance, jobStatus, task.getPreferredHost());
-          /**
-           *  A task store is active if all of the following conditions are true:
-           *  a) If the store is amongst the active stores of the task.
-           *  b) If the job has been started.
-           *  c) If the preferred host of the task is the localhost on which the monitor is run.
-           */
-          if (jobStatus.hasBeenStarted()
-              && task.getStoreNames().contains(storeName)
-              && task.getPreferredHost().equals(localHostName)) {
-            LOG.info(String.format("Store %s is actively used by the task: %s.", storeName, task.getTaskName()));
-          } else {
-            LOG.info(String.format("Store %s not used by the task: %s.", storeName, task.getTaskName()));
-            markSweepTaskStore(TaskStorageManager.getStorePartitionDir(jobDir, storeName, new TaskName(task.getTaskName())));
+      try {
+        JobStatus jobStatus = jobsClient.getJobStatus(jobInstance);
+        for (Task task : jobsClient.getTasks(jobInstance)) {
+          for (String storeName : jobDir.list(DirectoryFileFilter.DIRECTORY)) {
+            LOG.info("Job: {} has the running status: {} with preferred host:  {}", jobInstance, jobStatus, task.getPreferredHost());
+            /**
+             *  A task store is active if all of the following conditions are true:
+             *  a) If the store is amongst the active stores of the task.
+             *  b) If the job has been started.
+             *  c) If the preferred host of the task is the localhost on which the monitor is run.
+             */
+            if (jobStatus.hasBeenStarted()
+                && task.getStoreNames().contains(storeName)
+                && task.getPreferredHost().equals(localHostName)) {
+              LOG.info(String.format("Store %s is actively used by the task: %s.", storeName, task.getTaskName()));
+            } else {
+              LOG.info(String.format("Store %s not used by the task: %s.", storeName, task.getTaskName()));
+              markSweepTaskStore(TaskStorageManager.getStorePartitionDir(jobDir, storeName, new TaskName(task.getTaskName())));
+            }
           }
         }
+      } catch (Exception ex) {
+        if (!config.getIgnoreFailures()) {
+          throw ex;
+        }
+        LOG.warn("Local store cleanup for job: {} resulted in exception: {}. Config: {} turned on, failures are ignored.",
```

Comment:
> What's the scenario when this would and would not be turned on?

When failure in either getting status/task details of a job or  store cleanup of one job should not affect the other job store cleanup in NM host. Swallowing exceptions within code isn't usual modus operandi, hence felt the need that the client should control to say either stop at first failure or continue even when there're failures to next job(Making it explicit. This was problem within linkedin, since some jobs were installed in primary RM and not in standby RM, there by causing problems. In some cases, jobs were not installed in both the RM's). 

> Rephrase so that the exception is the last part of the message. 

Makes sense, Fixed it. Please check and see if it's alright.